src/Engine/Managers/ComponentMessenger/ComponentMessenger.inl

```diff
@@ -144,6 +144,7 @@ ComponentMessenger::registerInput( const Entity* entity,
                                    const std::string& id,
                                    const typename CallbackTypes<ReturnType>::Setter& cb ) {
     CORE_ASSERT( entity && comp->getEntity() == entity, "Component not added to entity" );
+    CORE_UNUSED( comp );
```

Comment:
I think this kind of macro is error prone. Instead of a macro, that expand to dead code while the name still remains usable after that, il might be safer to just omit the name of the parameter in the method/function definition.

Reply:
Hum, true.

Reply:
Well, not sure how it works, but do this change a lot of user code ?
If not, well I'm fine changing the interface right now.

Reply:
Nothing to change in the user code. Just give only type of parameter that is not used (without name). eg, for here (can't make a suggestion ...)
```cpp
ComponentMessenger::registerInput( const Entity* entity,
                                   Component* , // This un-named parameter can't be used in the method and will not generate "unused" warning
                                   const std::string& id,
                                   ...
```

Reply:
ah, I though of removing the comp parameter ;)